ur_robot_driver/src/ur/ur_driver.cpp

```diff
@@ -258,21 +258,25 @@ rtde_interface::RTDEWriter& UrDriver::getRTDEWriter()
   return rtde_client_->getWriter();
 }
 
-bool UrDriver::sendScript(const std::string& program)
+bool UrDriver::sendScript(std::string program)
 {
   if (secondary_stream_ == nullptr)
   {
     throw std::runtime_error("Sending script to robot requested while there is no primary interface established. This "
                              "should not happen.");
   }
 
-  size_t len = program.size();
   const uint8_t* data = reinterpret_cast<const uint8_t*>(program.c_str());
+  if (program.back() != '\n')
+  {
+    program.append("\n");
+  }
+  size_t len = program.size();
```

Comment:
Are scripts with empty lines not legal?

If they are, I would suggest something like this:

```c++
auto program_with_newline = program + '\n';
```
and then do whatever was done before.

Reply:
looks cleaner, I agree. Scripts with multiple empty lines should be working, as well. (Just tested to make sure)